src/components/Filter/Filter.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { useProduct } from '../../context/product-context'
+import "./Filter.css"
+function Filter() {
+    
+    const {state:{sortByPrice , sortByRating , sortByCategory:{mobile,laptop} , priceRange} ,dispatch} = useProduct()
```

Comment:
if you have destructured state in the context itself then here you can use values directly.

Reply:
will do